tests/func/test_remove.py

```diff
@@ -32,10 +32,40 @@ def test_remove(tmp_dir, scm, dvc, run_copy, remove_outs):
     assert not (tmp_dir / ".gitignore").exists()
 
 
+def test_remove_file_as_target(tmp_dir, scm, dvc):
+    dvc.run(
+        name="my",
+        cmd='echo "hello" > foo && echo "hello" > bar',
+        deps=[],
+        outs=["foo", "bar"],
+    )
+
+    dvc.remove("foo")
+    assert "/foo" not in get_gitignore_content()
+    assert "/bar" in get_gitignore_content()
```

Comment:
Maybe checking contents in `dvc.yaml` directly is better?

Reply:
I'm not sure if I fully understand your suggestion. As far as I know, using `dvc.remove("foo")` will not update the contents in `dvc.yaml`; it will remove `foo` from both the file system and the `.gitignore`

Reply:
This confused me, because according to [doc](https://dvc.org/doc/command-reference/remove#remove) `dvc remove`  delete stages or `.dvc` files, including removing `.dvc` file or removing stages in `dvc.yaml` and removing outputs from .gitignore. I don't know the exact behavior of this feature, and seems that it wasn't defined clearly in [documents](https://github.com/iterative/dvc.org/pull/2357#discussion_r608201613)

Reply:
I understand the source of confusion, we surely need to improve the documentation related with this change